followthemoney/graph.py

```diff
@@ -60,50 +72,59 @@ class Edge(object):
     __slots__ = ['id', 'weight', 'source_id', 'target_id',
                  'prop', 'proxy', 'schema', 'graph']
 
-    def __init__(self, graph, source, target, proxy=None, prop=None, value=None):  # noqa
+    def __init__(self, graph: 'Graph', source: Node, target: Node,
+                 proxy: Optional[EntityProxy]=None, prop: Optional[Property]=None,
+                 value: Optional[str]=None):  # noqa
         self.graph = graph
         self.id = f"{source.id}<>{target.id}"
         self.source_id = source.id
         self.target_id = target.id
         self.weight = 1.0
-        self.prop = prop
-        self.proxy = proxy
-        self.schema = None
+        self.prop: Optional[Property] = prop
+        self.proxy: Optional[EntityProxy] = proxy
+        self.schema: Optional[Schema] = None
         if prop is not None:
             self.weight = prop.specificity(value)
         if proxy is not None:
             self.id = f"{source.id}<{proxy.id}>{target.id}"
             self.schema = proxy.schema
 
     @property
-    def source(self):
+    def source(self) -> Optional[Node]:
```

Comment:
Not really optional.

Reply:
So should I do `self.graph.nodes.get[self.source_id]` instead?